util/simple_mixed_compressor.h

```diff
@@ -0,0 +1,131 @@
+    }
+    (void)dict;
+    (void)type;
+  }
+  size_t GetMaxSampleSizeIfWantDict(CacheEntryRole block_type) const override {
+    return compressors_.back()->GetMaxSampleSizeIfWantDict(block_type);
+  }
+
+  Slice GetSerializedDict() const override {
+    return compressors_.back()->GetSerializedDict();
+  }
+
+  CompressionType GetPreferredCompressionType() const override { return kZSTD; }
+
+  ManagedWorkingArea ObtainWorkingArea() override {
+    return compressors_.back()->ObtainWorkingArea();
+  }
+  virtual std::unique_ptr<Compressor> MaybeCloneSpecialized(
+      CacheEntryRole block_type, DictSampleArgs&& dict_samples) override {
+    return compressors_.back()->MaybeCloneSpecialized(block_type,
+                                                      std::move(dict_samples));
+  }
+
+ protected:
+  std::vector<std::unique_ptr<Compressor>> compressors_;
+
+ private:
+  mutable std::mutex mutex_;  // Protects access to current_index_
+};
+struct SimpleMixedCompressor : public MultiCompressorWrapper {
+  using MultiCompressorWrapper::MultiCompressorWrapper;
+  Status CompressBlock(Slice uncompressed_data, std::string* compressed_output,
+                       CompressionType* out_compression_type,
+                       ManagedWorkingArea* wa) override {
+    const auto& compressions = GetSupportedCompressions();
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(
+        1, (int)compressions.size() - 2);  // avoiding no compression and zstd
+    auto selected = dis(gen);
+    auto& compressor = compressors_[selected % compressors_.size()];
+    // fprintf(stdout, "[MultiCompressorWrapper] selected compressor
+    // typeint:%d\n",
+    //         selected);
+    Status status = compressor->CompressBlock(
+        uncompressed_data, compressed_output, out_compression_type, wa);
+    return status;
+  }
+};
+
+class SimpleMixedCompressionManager : public CompressionManagerWrapper {
+  using CompressionManagerWrapper::CompressionManagerWrapper;
+  const char* Name() const override { return wrapped_->Name(); }
+  std::unique_ptr<Compressor> GetCompressorForSST(
+      const FilterBuildingContext& context, const CompressionOptions& opts,
+      CompressionType preferred) override {
+    assert(preferred == kZSTD);
+    (void)context;
+    return std::make_unique<SimpleMixedCompressor>(opts, preferred);
+  }
+};
+
+struct RoundRobinCompressor : public MultiCompressorWrapper {
+  using MultiCompressorWrapper::MultiCompressorWrapper;
+  Status CompressBlock(Slice uncompressed_data, std::string* compressed_output,
+                       CompressionType* out_compression_type,
+                       ManagedWorkingArea* wa) override {
+    const auto& compressions = GetSupportedCompressions();
+    auto counter = block_counter.FetchAddRelaxed(1);
+    auto sel_idx = counter % (compressions.size() - 1);
+    auto type = compressions[sel_idx];
+    *out_compression_type = type;
```

Comment:
This line is unnecessary and misleading now. It is up to `compressor->CompressBlock` to populate `out_compression_type`